nvflare/lighter/poc_commands.py

```diff
@@ -78,57 +97,359 @@ def get_stop_cmd(poc_workspace: str, service_dir_name: str):
+
+
+def get_fl_client_names(project_config: OrderedDict) -> List[str]:
+    participants: List[dict] = project_config["participants"]
+    client_names = [p["name"] for p in participants if p["type"] == "client"]
+    return client_names
+
+
+def prepare_builders(project_dict: OrderedDict) -> List:
+    builders = list()
+    admin_name = [p["name"] for p in project_dict["participants"] if p["type"] == "admin"][0]
+    for b in project_dict.get("builders"):
+        path = b.get("path")
+        args = b.get("args")
+
+        if b.get("path") == "nvflare.lighter.impl.static_file.StaticFileBuilder":
+            path = "nvflare.lighter.impl.local_static_file.LocalStaticFileBuilder"
+            args["overseer_agent"]["args"]["sp_end_point"] = "localhost:8002:8003"
+            args["username"] = admin_name
+
+        elif b.get("path") == "nvflare.lighter.impl.cert.CertBuilder":
+            path = "nvflare.lighter.impl.local_cert.LocalCertBuilder"
+
+        builders.append(instantiate_class(path, args))
+    return builders
+
+
+def local_provision(
+    clients: List[str],
+    number_of_clients: int,
+    workspace: str,
+    docker_image: str,
+    use_he: bool = False,
+    project_conf_path: str = "",
+) -> Tuple:
+    user_provided_project_config = False
+    if project_conf_path:
+        src_project_file = project_conf_path
+        dst_project_file = os.path.join(workspace, "project.yml")
+        user_provided_project_config = True
+    else:
+        src_project_file = gen_project_config_file(workspace)
+        dst_project_file = src_project_file
+
+    print(f"provision at {workspace} for {number_of_clients} clients with {src_project_file}")
+    project_config: OrderedDict = load_yaml(src_project_file)
+    if not project_config:
+        raise CLIException(f"empty or invalid project config from project yaml file: {src_project_file}")
+
+    if not user_provided_project_config:
+        project_config = update_server_name(project_config)
+        project_config = update_clients(clients, number_of_clients, project_config)
+        project_config = add_he_builder(use_he, project_config)
+        if docker_image:
+            project_config = update_static_file_builder(docker_image, project_config)
+    save_project_config(project_config, dst_project_file)
+    packages = get_packages(project_config)
+    project = prepare_project(project_config)
+    builders = prepare_builders(project_config)
+    provisioner = Provisioner(workspace, builders)
+    provisioner.provision(project)
+
+    return project_config, packages
+
+
+def get_packages(project_config):
+    packages = {
+        SC.FLARE_SERVER: get_fl_server_name(project_config),
+        SC.FLARE_PROJ_ADMIN: get_proj_admin(project_config),
+        SC.FLARE_CLIENTS: get_fl_client_names(project_config),
+        SC.IS_DOCKER_RUN: is_docker_run(project_config),
+    }
+    return packages
```

Comment:
Should we call this get_service_config?
Or what does a "package" mean?

Reply:
Yes, for get_packages I would expect it might returning a list of directories, or a list of objects.

"global_packages" to me is also not that clear.
We actually mean "service_config"?

or call this method "get_service_config_from_project"?